### PR TITLE
Automatically sorted activity list and related tests

### PR DESCRIPTION
**Pull request by ivyyangyq**
More to be added.

### CODE REVIEW

**Comment by ivyyangyq** [comment]
in progress: Updated tests with TypicalActivities for activity commands
refactor TypicalPersons.getTypicalAddressBook to getTypicalAddressBookWithPersons
Activity Command Tests and UniqueActivityList test to be updated in later commits

File: src/main/java/seedu/address/model/activity/UniqueActivityList.java [hunk 1]
```diff
@@ -25,8 +26,9 @@
 public class UniqueActivityList implements Iterable<Activity> {
 
     private final ObservableList<Activity> internalList = FXCollections.observableArrayList();
+    private final SortedList<Activity> sortedInternalList = internalList.sorted();
```
Comment (minernchan) [inline src/main/java/seedu/address/model/activity/UniqueActivityList.java:29]: When the activity list is displayed, only the internalUnmodifiableList (sorted) is displayed? We will still retain the original internalList with the original order that the activities were added into the address book?

File: src/main/java/seedu/address/model/activity/Activity.java [hunk 2]
```diff
@@ -136,4 +167,18 @@ public String toString() {
                 .append(getNumberAttending());
         return builder.toString();
     }
+
+    @Override
+    public int compareTo(Activity other) {
+        //when both activity are ongoing or completed, compare by the time of the activity
+        if (this.status.equals(other.status)) {
```
Comment (minernchan) [inline src/main/java/seedu/address/model/activity/Activity.java]: Should use getStatus() instead of .status

**Review by minernchan — APPROVED** [review]

File: src/main/java/seedu/address/model/activity/Activity.java (hunk 2 above)
Comment (ivyyangyq) [reply src/main/java/seedu/address/model/activity/Activity.java]: okay!

**Pull request merged**